ports/winit/app.rs

```diff
@@ -346,12 +346,27 @@ fn get_default_url() -> ServoUrl {
     // If the url is not provided, we fallback to the homepage in prefs,
     // or a blank page in case the homepage is not set either.
     let cwd = env::current_dir().unwrap();
-    let cmdline_url = opts::get().url.clone();
+    let mut cmdline_url = opts::get().url.clone();
     let pref_url = {
         let homepage_url = pref!(shell.homepage);
         parse_url_or_filename(&cwd, &homepage_url).ok()
     };
     let blank_url = ServoUrl::parse("about:blank").ok();
 
+    // Check if a command-line URL is provided
+    if let Some(url) = cmdline_url.clone() {
+        let url_path = url.path();
+
+        // Check if the URL path corresponds to a file
+        if fs::metadata(url_path).is_ok() && fs::metadata(url_path).unwrap().is_file() {
+            return cmdline_url.unwrap();
+        }
+
+        // Extract domain from URL path
+        let domain = url_path.split('/').last().unwrap_or("");
```

Comment:
Maybe we can convert this into an actual URL object in order to get the domain? Looking for slashes and such ourselves without actually trying to parse a URL seems a little brittle.